File: fixtures/python/hardcoded_security_constant/negative.py
# SEC012-hardcoded-security-constant: negative fixture
# None of the following should produce findings.

import os

# RHS is a call expression (environment variable lookup)
password = os.environ["PASSWORD"]
api_key = os.getenv("API_KEY")

# Excluded suffixes
total_password_count = 0
secret_handler = None
token_type = "bearer"
auth_name = "admin"

# Empty string
password_hash = ""

# Unrelated names
username = "admin"
host = "localhost"

File: fixtures/js/hardcoded_security_constant/negative.ts
// SEC012-hardcoded-security-constant: negative fixture
// None of the following should produce findings.

// RHS is a member expression (environment variable lookup)
const api_key = process.env.API_KEY;
const password = process.env.PASSWORD;

// Excluded suffixes
const total_password_count = 0;
const secret_handler = new Object();
const token_type = "bearer";

// Empty string
const passwordHash = "";

// Unrelated names
const username = "admin";
const host = "localhost";
